pkg/morserino_com/listen_com.go

```diff
@@ -24,67 +24,126 @@ THE SOFTWARE.
 
 import (
 	"fmt"
+	"io"
 	"log"
 	"strings"
+	"testing/iotest"
 
-	"morserino_display/pkg/morserino_console"
-
+	"github.com/on4kjm/morserino_display/pkg/morserino_console"
 	"go.bug.st/serial"
 )
 
+// Channel used to forward data received on the serial port to the display modules
+var MessageBuffer = make(chan string, 10)
+
 // Main listen function with display to the console
 func Listen_console(morserinoPortName string, genericEnumPorts comPortEnumerator) error {
 
-	//Port parameters for a Morserino
-	mode := &serial.Mode{
-		BaudRate: 115200,
-		Parity:   serial.NoParity,
-		DataBits: 8,
-		StopBits: serial.OneStopBit,
+	//If requested, use the simulator instead of a real Morserino
+	if strings.HasPrefix("SIMULATOR", strings.ToUpper(morserinoPortName)) {
+		TestMessage := "cq cq de on4kjm on4kjm = tks fer call om = ur rst 599 = hw? \n73 de on4kjm = <sk> e e"
+		Listen(iotest.OneByteReader(strings.NewReader(TestMessage)))
+		return nil
 	}
 
 	//If portname "auto" was specified, we scan for the Morserino port
 	if strings.ToUpper(morserinoPortName) == "AUTO" {
-		theComPortList, err := Get_com_list(genericEnumPorts)
+		portName, err := Get_MorserinoPort_automatically(genericEnumPorts)
 		if err != nil {
 			return err
 		}
+		morserinoPortName = portName
+	}
 
-		if theComPortList.nbrOfMorserinoPorts == 0 {
-			fmt.Println("Didn't find a connected Morserino!")
-			return fmt.Errorf("Did not find a usable port.")
-		}
-
-		if theComPortList.nbrOfMorserinoPorts > 1 {
-			fmt.Println("ERROR: Multiple Morserino devices found.")
-			return fmt.Errorf("ERROR: Multiple Morserino devices found.")
-		}
+	log.Println("Listening to port \"" + morserinoPortName + "\"")
 
-		morserinoPortName = theComPortList.morserinoPortName
-		fmt.Println("Automatically detected Morserino port: " + morserinoPortName + "\n")
+	//Port parameters for a Morserino
+	mode := &serial.Mode{
+		BaudRate: 115200,
+		Parity:   serial.NoParity,
+		DataBits: 8,
+		StopBits: serial.OneStopBit,
 	}
 
-	fmt.Println("Listening to port \"" + morserinoPortName + "\"")
-	myPort, err := genericOpenComPort.Open(morserinoPortName, mode)
+	p, err := serial.Open(morserinoPortName, mode)
 	if err != nil {
-		log.Fatal(err)
+		return err
 	}
+	defer p.Close()
 
+	return Listen(p)
+}
+
+// Main receive loop
+func Listen(port io.Reader) error {
+	//TODO: needs to be moved as a goroutine
 	consoleDisplay := morserino_console.ConsoleDisplay{}
 
+	var closeKey string
+	possibleExitRequest := false
+	closeRequested := false
```

Comment:
```suggestion
	var (
		closeKey string
		possibleExitRequest bool
		closeRequested bool
	)
```

nit: zero value of a boolean is false, so you  don't need to intialize your boolean variable to false as it already is :)

Reply:
Thanks for that tip (and for showing the multi-var declaration format)